Sources/Turf/Geometries/LineString.swift

```diff
@@ -78,9 +78,63 @@ extension LineString {
     }
     
     /**
-     Returns the portion of the line string that begins at the given coordinate and extends the given distance along the line string.
+     Returns the portion of the line string that begins at the given start distance and extends the given stop distance along the line string.
      
-     This method is roughly equivalent to the [turf-line-slice-along](https://turfjs.org/docs/#lineSliceAlong) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-slice-along/)), except that it accepts a starting position instead of a starting distance along the line string.
+     This method is roughly equivalent to the [turf-line-slice-along](https://turfjs.org/docs/#lineSliceAlong) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-slice-along/)).
+     */
+    public func lineSliceAlong(startDistance: LocationDistance, stopDistance: LocationDistance) -> LineString? {
+        let coordinates = self.coordinates
+        var traveled: LocationDistance = 0
+        var slice = [LocationCoordinate2D]()
+        
+        for i in 0..<coordinates.endIndex {
```

Comment:
This is a much more literal port of the JavaScript method than the existing method below. Do we know if the version below was correct, apart from the difference in inputs? We should find a way for one method to be based on the other.

Reply:
The method below is different from the `trimmed(from:to:)` because it choose a `closestCoordinate` as a start point instead of a start distance. If we convert the `closestCoordinate` to the start distance and then use the `trimmed(from:to:)`, it would lead to more calculation than the original one. 

And I tried to use the below method to replace the `trimmed(from:to:)` , but it could not pass some tests. And the `trimmed(from:to:)` has much less calculation than the twice use of `trimmed(from:distance:)`. 

So I think they two are better kept separate for different use cases.